src/rnn_api.cpp

```diff
@@ -36,6 +36,62 @@ enum RNNDir_t
     BackwardData,
     BackwardWeights
 };
+
+extern "C" miopenStatus_t
+miopenSetRNNDataSeqTensorDescriptor(miopenSeqTensorDescriptor_t seqTensorDesc,
+                                    miopenDataType_t dataType,
+                                    miopenRNNBaseLayout_t layout,
+                                    int maxSequenceLen,
+                                    int batchSize,
+                                    int vectorSize,
+                                    const int* sequenceLenArray,
+                                    void* paddingMarker)
+{
+    (void)paddingMarker;
+    MIOPEN_LOG_FUNCTION(
+        seqTensorDesc, dataType, layout, maxSequenceLen, batchSize, vectorSize, sequenceLenArray);
+
+    return miopen::try_([&] {
+        miopen::deref(seqTensorDesc) = miopen::RNNDescriptor::makeSeqTensorDescriptor(
+            dataType, layout, maxSequenceLen, batchSize, vectorSize, sequenceLenArray);
+    });
+}
+
+extern "C" miopenStatus_t
+miopenGetRNNDataSeqTensorDescriptor(miopenSeqTensorDescriptor_t seqTensorDesc,
+                                    miopenDataType_t* dataType,
+                                    miopenRNNBaseLayout_t* layout,
+                                    int* maxSequenceLen,
+                                    int* batchSize,
+                                    int* vectorSize,
+                                    int sequenceLenArrayLimit,
+                                    int* sequenceLenArray,
+                                    void* paddingMarker)
+{
+    (void)paddingMarker;
+    MIOPEN_LOG_FUNCTION(seqTensorDesc, sequenceLenArrayLimit);
+
+    const auto status = miopen::try_([&] {
+        *dataType = miopen::deref(seqTensorDesc).GetType();
+        *layout = miopen::RNNDescriptor::getBaseLayoutFromDataTensor(miopen::deref(seqTensorDesc));
+
+        miopen::tie_deref(batchSize, maxSequenceLen, vectorSize) =
+            miopen::tien<3>(miopen::deref(seqTensorDesc).GetLengths());
+
+        if(sequenceLenArray != nullptr)
+        {
+            size_t copy_sz =
+                std::min(static_cast<size_t>(sequenceLenArrayLimit),
+                         miopen::deref(seqTensorDesc).GetSequenceLengthsVector().size());
+            const auto seq_lens_begin =
+                miopen::deref(seqTensorDesc).GetSequenceLengthsVector().begin();
+            std::copy(seq_lens_begin, seq_lens_begin + copy_sz, sequenceLenArray);
```

Comment:
Seems `std::copy_n` is better.
```suggestion
            std::copy_n(seq_lens_begin, copy_sz, sequenceLenArray);
```